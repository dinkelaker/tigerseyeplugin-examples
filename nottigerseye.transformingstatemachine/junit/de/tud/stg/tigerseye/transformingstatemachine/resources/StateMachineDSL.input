	StateMachine m = machine Watch {
			start state reseted {
				entry: output "entering reseted";
		 		exit: output "leaving reseted";
				 transitions {
				 	
				 	 start → running;
				 	 
				     if (variabilityFlag) { 
  					   switchOff → off;
  					 } else {
  					   switchOff → paused;  					 
  					 } 
				 }
			 }
								                            
			 state running {
				entry: output "entering running";
			 	exit: output "leaving running";
				 transitions {
					 split → paused;
					 stop → stopped;
				 }
			 }
								                            
			 state paused {
				entry: output "entering paused";
			 	exit: output "leaving paused";
				 transitions {
					 unsplit → running;
					 stop → Stopped;
				 }
			 }
								                           
			 state stopped {
				entry: output "entering stopped";
			 	exit: output "leaving stopped";
				 transitions {
					 reset → stopped;
					 switchOff → off;
				 }
			 }
								                            
			 state off {
				 entry: output "entering off";
				 exit: output "leaving off";
				 transitions {
					 toEnd → reseted;
				 }	
			 }
		}
			
	m.start();
	m.input("start");
	m.input("split");
	m.input("unsplit");
	m.input("stop");
	m.input("switchOff");
	m.input("toEnd");