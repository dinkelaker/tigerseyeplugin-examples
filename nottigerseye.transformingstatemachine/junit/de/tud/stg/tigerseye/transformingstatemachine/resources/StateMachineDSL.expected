	StateMachine m = machine__p0_lcub_p1_rcub(
		"Watch",
		[
			start__state__p0_lcub_p1__p2_rcub(
				"reseted",
				[
					entry_colon_p0_semi(
						output__p0(
							"entering reseted")
					),
					exit_colon_p0_semi(
						output__p0(
							"leaving reseted")
					)
				] as ActionType[],
				transitions_lcub_p0_rcub(
					{
						p0_rarr_p1_semi(
							"start",
							"running")
						if (variabilityFlag) {
							p0_rarr_p1_semi(
								"switchOff",
								"off")
						}
						else {
							p0_rarr_p1_semi(
								"switchOff",
								"paused")
						}
					}
				)
			),
			state__p0_lcub_p1__p2_rcub(
				"running",
				[
					entry_colon_p0_semi(
						output__p0(
							"entering running")
					),
					exit_colon_p0_semi(
						output__p0(
							"leaving running")
					)
				] as ActionType[],
				transitions_lcub_p0_rcub(
					{
						p0_rarr_p1_semi(
							"split",
							"paused")
						p0_rarr_p1_semi(
							"stop",
							"stopped")
					}
				)
			),
			state__p0_lcub_p1__p2_rcub(
				"paused",
				[
					entry_colon_p0_semi(
						output__p0(
							"entering paused")
					),
					exit_colon_p0_semi(
						output__p0(
							"leaving paused")
					)
				] as ActionType[],
				transitions_lcub_p0_rcub(
					{
						p0_rarr_p1_semi(
							"unsplit",
							"running")
						p0_rarr_p1_semi(
							"stop",
							"Stopped")
					}
				)
			),
			state__p0_lcub_p1__p2_rcub(
				"stopped",
				[
					entry_colon_p0_semi(
						output__p0(
							"entering stopped")
					),
					exit_colon_p0_semi(
						output__p0(
							"leaving stopped")
					)
				] as ActionType[],
				transitions_lcub_p0_rcub(
					{
						p0_rarr_p1_semi(
							"reset",
							"stopped")
						p0_rarr_p1_semi(
							"switchOff",
							"off")
					}
				)
			),
			state__p0_lcub_p1__p2_rcub(
				"off",
				[
					entry_colon_p0_semi(
						output__p0(
							"entering off")
					),
					exit_colon_p0_semi(
						output__p0(
							"leaving off")
					)
				] as ActionType[],
				transitions_lcub_p0_rcub(
					{
						p0_rarr_p1_semi(
							"toEnd",
							"reseted")
					}
				)
			)
		] as State[]
	)
	
	m.start ();
	m.input ("start");
	m.input ("split");
	m.input ("unsplit");
	m.input ("stop");
	m.input ("switchOff");
	m.input ("toEnd");